cmake/Dependencies.cmake

```diff
@@ -1169,6 +1169,49 @@ if(USE_GLOO)
   endif()
 endif()
 
+if(USE_TENSORPIPE)
+  if(MSVC)
+    message(WARNING "Tensorpipe cannot be used on Windows.")
+  else()
+    if(CAFFE2_LINK_LOCAL_PROTOBUF)
+      set(TP_EXTRA_CMAKE_ARGS -DProtobuf_DIR=${PROJECT_BINARY_DIR}/third_party/protobuf/cmake/${CMAKE_INSTALL_LIBDIR}/cmake/protobuf)
+    endif()
+
+    include(ExternalProject)
+
+    ExternalProject_Add(cmake-3.17.1
```

Comment:
Is there any reason that we need to download a specific version of cmake here? Can we directly using the same cmake the rest of the repo is using?

Reply:
Sadly we need a more recent version than the pytorch minimum pinned one (which is 3.5), among other things for protobufs. An other option would be to bump pytorch's minimum cmake version (not sure what the constraints are on this, though it is simply a build requirement), but I wanted to make this PR as little invasive as possible.

Reply:
@ezyang @malfet @seemethere is it OK to download a different version of cmake here?

Reply:
This seems REAAALLY questionable XD

Reply:
@beauby what version of protobuf do you use? pytroch is using 3.11.4, which seems to be compatible with cmake-3.5